aiida/orm/nodes/data/structure.py

```diff
@@ -1203,7 +1203,7 @@ def get_site_kindnames(self):
         """
         return [this_site.kind_name for this_site in self.sites]
 
-    def get_composition(self):
+    def get_composition(self, type):
```

Comment:
`type` is a protected keyword in Python, so it is best not to override it with this function argument. A better option would be `mode` perhaps.

Also, when you add an argument. Please add a docstring to say what it does, e.g.:
```
:param mode: Specify the mode of the composition to return. Choose from ``full``, ``reduced`` or ``fractional``. For ``reduced`` mode, the counts are renormalized to the greatest common denominator. For ``fractional`` mode, the counts are renormalized such that the sum equals 1.

Reply:
Also, you should add a default for this new argument, because now you are breaking backwards compatibility. If you are now calling `get_composition()` as one used to, an exception will be raised. The default should be the value of the old-behavior such that invoking it without argument `get_composition()` still works the same.